docusaurus/docs/Android/03-ui/04-components/03-message-list.md

```diff
@@ -1,14 +1,63 @@
 # Message List
 
-<!-- TODO: Import whatever makes sense to import from https://getstream.io/chat/docs/android/message_list_view/?language=kotlin -->
+`MessageListView` is one of our core UI components. Generally speaking, it's a list of messages for some particular channel. The `MessageListView` contains the following list of possible child items:
 
-## Overview
+- Plain text message
+- Text and attachments (media or file) message
+- Deleted message (only for current user)
+- Error message (e.g. autoblocked message with inappropriate content)
+- System message (e.g. some user joined to a channel)
+- Giphy preview
+- Date separator
+- Loading more indicator
+- Thread separator (for thread mode only)
+- Typing indicator
 
-<!-- TODO: Brief description and a couple screenshots with default styling. -->
+Using custom attributes and methods in runtime you're able to customize the appearance of this component. Also `MessageListView` contains the set of overridable action/option handlers and event listeners. By default this component has the following look:
+
+| Light Mode | Dark Mode |
+| --- | --- |
+|![Message list overview in light mode](../../assets/message_list_view_overview_light.png)|![Message list overview in dark mode](../../assets/message_list_view_overview_dark.png)|
+
+## Usage
+If you want to use all default features and default design of this component then start is easy. It consists of two steps:
+1. Adding component to your xml layout hierarchy
+2. Bind out-of-box `MessageListViewModel` and `MessageLisView`.
+
+### Adding to xml layout
+Adding `MessageListView` to your layout is easy as inserting following lines to your layout hierarchy (example for `ConstraintLayout`):
+
+```xml
+<io.getstream.chat.android.ui.message.list.MessageListView
+        android:id="@+id/message_list_view"
+        android:layout_width="0dp"
+        android:layout_height="0dp"
+        app:layout_constraintBottom_toBottomOf="parent"
+        app:layout_constraintLeft_toLeftOf="parent"
+        app:layout_constraintRight_toRightOf="parent"
+        app:layout_constraintTop_toTopOf="parent"
+        />
+``` 
+
+### ViewModel and binding
+UI components provide out-of-box a view model for `MessageListView` and the `bindView` extension function that makes default setup:
+
+```kotlin
+class MessageListFragment : Fragment() {
+
+    private lateinit var messageListView: MessageListView
+
+    fun bindView() {
+        val channelCID = "messaging:123"
+        val viewModel = MessageListViewModel(cid = channelCID)
+        viewModel.bindView(messageListView, this)
```

Comment:
We should showcase using the factory instead of creating a ViewModel directly, otherwise clients are gonna copy this code and not have their ViewModels retained. We can probably also drop the Fragment context of this code snippet (especially since we're not initializing the `lateinit var` anyway) - see what the Channel List docs do for both of these.